src/main/java/com/hpe/cloudfoundryjenkins/CloudFoundryPushPublisher.java

```diff
@@ -337,42 +337,78 @@ private boolean processOneApp(CloudFoundryClient client, DeploymentInfo deployme
 
     private boolean createApplicationIfNeeded(CloudFoundryClient client, BuildListener listener,
                                               DeploymentInfo deploymentInfo, String appURI) {
-        // Check if app already exists
-        List<CloudApplication> existingApps = client.getApplications();
-        boolean createNewApp = true;
-        for (CloudApplication app : existingApps) {
-            if (app.getName().equals(deploymentInfo.getAppName())) {
-                if (resetIfExists) {
-                    listener.getLogger().println("App already exists, resetting.");
-                    client.deleteApplication(deploymentInfo.getAppName());
-                    listener.getLogger().println("App deleted.");
-                } else {
-                    createNewApp = false;
-                    listener.getLogger().println("App already exists, skipping creation.");
-                }
-                break;
-            }
+        // Get app instance
+        CloudApplication cloudApplication = client.getApplication(deploymentInfo.getAppName());
```

Comment:
Unfortunately this causes a CloudFoundryException if the app doesn't already exist. Either go through getApplications() like I did, or try catching the exception.